messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -237,6 +227,53 @@ protected void createPartitionReaders() throws IOException {
         messageConsumedMap.put(id, false);
       }
     }
+    initDone = true;
+  }
+
+  private void createPartitionReader(String clusterName, String collector, Map<PartitionId,
+      PartitionCheckpoint> partitionsChkPoints,String fsuri, FileSystem fs,
+      Path streamDir, String defaultClusterName, Path rootdir) throws IOException {
+
+    PartitionId id = new PartitionId(clusterName, collector);
+    PartitionCheckpoint pck = partitionsChkPoints.get(id);
+    /*
+    * Migration of checkpoint required in this case
+    * If user provides a cluster name and partition checkpoint is null
+    */
+    if (!clusterName.equals(defaultClusterName) && pck == null) {
+      PartitionId defaultPid = new PartitionId(defaultClusterName,
+          collector);
+      pck = partitionsChkPoints.get(defaultPid);
+      /**
+       * Migrate to new checkpoint
+       */
+      ((Checkpoint) currentCheckpoint).migrateCheckpoint(pck, defaultPid, id);
+    }
+    Date partitionTimestamp = getPartitionTimestamp(id, pck);
+    LOG.debug("Creating partition " + id);
+    PartitionReaderStatsExposer collectorMetrics = new
+        CollectorReaderStatsExposer(topicName, consumerName,
+        id.toString(), consumerNumber, fsuri);
+    addStatsExposer(collectorMetrics);
+    Path streamsLocalDir = null;
+    if (readFromLocalStream) {
+      streamsLocalDir = DatabusUtil.getStreamDir(StreamType.LOCAL,
+          rootdir, topicName);
+    }
+    for (int c = 0; c < numList; c++) {
+      collectorMetrics.incrementListOps();
+    }
+    PartitionReader newReader = new PartitionReader(id, pck, conf, fs,
+        new Path(streamDir, collector), streamsLocalDir, buffer, topicName,
+        partitionTimestamp, waitTimeForFlush, waitTimeForFileCreate,
+        collectorMetrics, stopTime);
+    if (!initDone) {
```

Comment:
let's have a comment specifying that timer gets triggered post-initialization, and hence timer will add in the newReaders map.